trait bound not satisfied "sealed trait" detection is too eager

### Code

```Rust
mod x {
    pub trait A {}
    pub trait B: A {}
    
    pub struct C;
    impl B for C {}
}
```

### Current output

```Shell
error[E0277]: the trait bound `C: A` is not satisfied
 --> src/lib.rs:6:16
  |
6 |     impl B for C {}
  |                ^ the trait `A` is not implemented for `C`
  |
help: this trait has no implementations, consider adding one
 --> src/lib.rs:2:5
  |
2 |     pub trait A {}
  |     ^^^^^^^^^^^
note: required by a bound in `B`
 --> src/lib.rs:3:18
  |
3 |     pub trait B: A {}
  |                  ^ required by this bound in `B`
  = note: `B` is a "sealed trait", because to implement it you also need to implement `x::A`, which is not accessible; this is usually done to force you to use one of the provided types that already implement it

For more information about this error, try `rustc --explain E0277`.
```

### Desired output

```Shell
error[E0277]: the trait bound `C: A` is not satisfied
 --> src/lib.rs:6:16
  |
6 |     impl B for C {}
  |                ^ the trait `A` is not implemented for `C`
  |
help: this trait has no implementations, consider adding one
 --> src/lib.rs:2:5
  |
2 |     pub trait A {}
  |     ^^^^^^^^^^^
note: required by a bound in `B`
 --> src/lib.rs:3:18
  |
3 |     pub trait B: A {}
  |                  ^ required by this bound in `B`

For more information about this error, try `rustc --explain E0277`.
```

### Rationale and extra context

In this case, the note does not apply, because in the context where we `impl B for C`, we could also `impl A for C`, so the trait may match the pattern of a sealed trait, but it is not sealed to us in this context.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-unknown-linux-gnu
release: 1.88.0
LLVM version: 20.1.5
```

### Anything else?

Seems this implementation was too eager: https://github.com/rust-lang/rust/commit/717c4817391f61953eed678607168cfea81e3111

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/tests/ui/privacy/sealed-traits/false-sealed-traits-note.rs
+++ b/tests/ui/privacy/sealed-traits/false-sealed-traits-note.rs
@@ -1,5 +1,6 @@
-// We should not emit sealed traits note, see issue #143392
+// We should not emit sealed traits note, see issue #143392 and #143121
 
+/// Reported in #143392
 mod inner {
     pub trait TraitA {}
 
@@ -9,5 +10,14 @@ mod inner {
 struct Struct;
 
 impl inner::TraitB for Struct {} //~ ERROR the trait bound `Struct: TraitA` is not satisfied [E0277]
+
+/// Reported in #143121
+mod x {
+    pub trait A {}
+    pub trait B: A {}
+
+    pub struct C;
+    impl B for C {} //~ ERROR the trait bound `C: A` is not satisfied [E0277]
+}
 
 fn main(){}

--- a/tests/ui/privacy/sealed-traits/false-sealed-traits-note.stderr
+++ b/tests/ui/privacy/sealed-traits/false-sealed-traits-note.stderr
@@ -1,20 +1,37 @@
 error[E0277]: the trait bound `Struct: TraitA` is not satisfied
-  --> $DIR/false-sealed-traits-note.rs:11:24
+  --> $DIR/false-sealed-traits-note.rs:12:24
    |
 LL | impl inner::TraitB for Struct {}
    |                        ^^^^^^ the trait `TraitA` is not implemented for `Struct`
    |
 help: this trait has no implementations, consider adding one
-  --> $DIR/false-sealed-traits-note.rs:4:5
+  --> $DIR/false-sealed-traits-note.rs:5:5
    |
 LL |     pub trait TraitA {}
    |     ^^^^^^^^^^^^^^^^
 note: required by a bound in `TraitB`
-  --> $DIR/false-sealed-traits-note.rs:6:23
+  --> $DIR/false-sealed-traits-note.rs:7:23
    |
 LL |     pub trait TraitB: TraitA {}
    |                       ^^^^^^ required by this bound in `TraitB`
 
-error: aborting due to 1 previous error
+error[E0277]: the trait bound `C: A` is not satisfied
+  --> $DIR/false-sealed-traits-note.rs:20:16
+   |
+LL |     impl B for C {}
+   |                ^ the trait `A` is not implemented for `C`
+   |
+help: this trait has no implementations, consider adding one
+  --> $DIR/false-sealed-traits-note.rs:16:5
+   |
+LL |     pub trait A {}
+   |     ^^^^^^^^^^^
+note: required by a bound in `B`
+  --> $DIR/false-sealed-traits-note.rs:17:18
+   |
+LL |     pub trait B: A {}
+   |                  ^ required by this bound in `B`
+
+error: aborting due to 2 previous errors
 
 For more information about this error, try `rustc --explain E0277`.

